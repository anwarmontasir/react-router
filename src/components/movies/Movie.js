import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './Movie.css';

export default class Movie extends Component {
  
  static propTypes = {
    imdbID: PropTypes.string,
    Poster: PropTypes.string,
    Title: PropTypes.string,
    Year: PropTypes.string
  };

  render() {
    const { imdbID, Poster, Title, Year } = this.props;

    return (
      <li className={styles.movie}>
        <Link to={`/movies/${imdbID}`}>
          <img alt={Title} src={Poster}/>
          <h3>{Title}</h3>
          <p>Released {Year}</p>
        </Link>
      </li>
    );
  }
}