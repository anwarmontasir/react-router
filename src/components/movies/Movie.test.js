import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Movie from './Movie';

describe('Movie', () => {
  it('renders with props', () => {
    const Poster = 'https://images-na.ssl-images-amazon.com/images/M/MV5BNzVlY2MwMjktM2E4OS00Y2Y3LWE3ZjctYzhkZGM3YzA1ZWM2XkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg';
    const wrapper = shallow(<Movie Title="Star Wars" Year="1977" Poster={Poster}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});