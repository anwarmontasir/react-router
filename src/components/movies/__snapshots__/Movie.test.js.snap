// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie renders with props 1`] = `
<li>
  <Link
    replace={false}
    to="/movies/undefined"
  >
    <img
      alt="Star Wars"
      src="https://images-na.ssl-images-amazon.com/images/M/MV5BNzVlY2MwMjktM2E4OS00Y2Y3LWE3ZjctYzhkZGM3YzA1ZWM2XkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg"
    />
    <h3>
      Star Wars
    </h3>
    <p>
      Released 
      1977
    </p>
  </Link>
</li>
`;
